Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugAuditTrail.cpp

```diff
@@ -0,0 +1,251 @@
+                                            childElem->GetName().c_str(),
+                                            (ImGuiTreeNodeFlags_Leaf | ImGuiTreeNodeFlags_NoTreePushOnOpen |
+                                             ImGuiTreeNodeFlags_SpanFullWidth));
+                                        ImGui::TableNextColumn();
+                                        ImGui::TableNextColumn();
+                                        ImGui::TableNextColumn();
+                                        ImGui::Text("%s", cliServValues.first.c_str());
+                                        ImGui::TableNextColumn();
+                                        ImGui::Text("%s", cliServValues.second.c_str());
+                                    }
+                                    ImGui::TreePop();
+                                }
+                            }
+                            else
+                            {
+                                ImGui::Text(child.m_name.c_str());
+                                ImGui::TableNextColumn();
+                                ImGui::TableNextColumn();
+                                ImGui::TableNextColumn();
+                                ImGui::TableNextColumn();
+                            }
+                        }
+                        if (elem->m_category != AuditCategory::Desync)
+                        {
+                            ImGui::TreePop();
+                        }
+                    }
+                    else
+                    {
+                        ImGui::TableNextColumn();
+                        ImGui::Text("%d", elem->m_inputId);
+                        ImGui::TableNextColumn();
+                        ImGui::Text("%d", elem->m_hostFrameId);
+                        ImGui::TableNextColumn();
+                        ImGui::TableNextColumn();
+                        ImGui::TableNextRow();
+                    }
+                }
+            }
+
+            if (!atRootLevel)
+            {
+                // Make sure to pop back to root on the way out
+                ImGui::TreePop();
+            }
+
+            ImGui::EndTable();
+            ImGui::NewLine();
+        }
+        ImGui::EndChild();
+#endif
+    }
+
+    void MultiplayerDebugAuditTrail::UpdateDebugOverlay()
+    {
+        if (m_debugDisplay == nullptr)
+        {
+            AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+            AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, AzFramework::g_defaultSceneEntityDebugDisplayId);
+            m_debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+        }
+
+        const AZ::u32 stateBefore = m_debugDisplay->GetState();
+        m_debugDisplay->SetColor(AZ::Colors::White);
+
+        m_debugDisplay->SetState(stateBefore);
+    }
+
+    bool MultiplayerDebugAuditTrail::CanPumpAuditTrail()
+    {
+        bool canPump = m_canPumpTrail;
```

Comment:
nit: Not too sure why this changes state. Can<I do something> seems like it should not change internal state.